airflow/www/auth.py

```diff
@@ -37,7 +37,11 @@ def decorated(*args, **kwargs):
             appbuilder = current_app.appbuilder
 
             dag_id = (
-                request.args.get("dag_id") or request.form.get("dag_id") or (request.json or {}).get("dag_id")
+                request.args.get("dag_id")
+                or request.form.get("dag_id")
+                or request.is_json
+                and request.json.get("dag_id")
+                or None
             )
```

Comment:
Let’s add parentheses around `request.is_json and request.json.get("dag_id")` to make this easier to understand

Reply:
Yeah. Readability is king